### ISSUE TITLE
The url-launcher-web after update 2.0.7 is causing issue. 

### ISSUE THREAD

**Issue by DakshDB** [flutter/flutter#97796]
## Steps to Reproduce

1. Use url-launcer-web 2.0.7 version.
2. Issue while building

**Expected results:** 

**Actual results:** 

../../Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_web-2.0.7/lib/url_launcher_web.dart:50:61: Error: No named parameter with the name 'isVisible'.
        .registerViewFactory(linkViewType, linkViewFactory, isVisible: false);

<details>
<summary>Code sample</summary>

```dart
```

</details>

<details>
  <summary>Logs</summary>

Launching lib\main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...
../../Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher_web-2.0.7/lib/url_launcher_web.dart:50:61: Error: No named parameter with the name 'isVisible'.
        .registerViewFactory(linkViewType, linkViewFactory, isVisible: false);
                                                            ^^^^^^^^^
Failed to compile application.

```
```

```
```

```
```

</details>

**Comment by darshankawar** [comment]
@DakshDB 
Can you provide `flutter doctor -v`, `pubspec.yaml` and [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) in properly formatted text that shows this issue ?

**Comment by Bilonik** [comment]
Same here, 
Use url_launcher_web: 2.0.6

**Comment by DakshDB** [comment]
- **flutter doctor -v** 

```
[√] Flutter (Channel stable, 2.8.1, on Microsoft Windows [Version 10.0.22000.434], locale en-IN)
    • Flutter version 2.8.1 at C:\Users\daksh\Documents\Flutter\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (7 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\daksh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

[√] VS Code (version 1.64.0)
    • VS Code at C:\Users\daksh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 97.0.4692.99
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 97.0.1072.76

• No issues found!
```

**- Pubspec.yaml**

```
name: portfolio
description: DB Portfolio

publish_to: 'none' # Remove this line if you wish to publish to pub.dev


version: 1.0.0+1

environment:
  sdk: ">=2.15.1 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^1.0.0
  url_launcher: ^6.0.4
  font_awesome_flutter: ^9.2.0


flutter:

  uses-material-design: true

```

- **Code**
```
import 'package:url_launcher/url_launcher.dart';

class Method {
  launchURL(String link) async {
    var url = link;
    if (await canLaunch(url)) {
      await launch(url);
    } else {
      throw 'Could not launch $url';
    }
  }
}
```

The error occurs while building for web

**Comment by DakshDB** [comment]
> Same here, Use url_launcher_web: 2.0.6

That would work, but using url_launcher for cross-platform support fails, due to dependency on url_launcher_web 2.0.7

**Comment by stuartmorgan** [comment]
2.0.7 requires requires Flutter 2.10, but that accidentally wasn't included in the pubspec requirements. It has been retracted to address this; 2.0.8 will be released later with the correct minimum dependency.

**Comment by c-lucera-pvotal** [comment]
This is still happening with a newly created project, **url_launcher: 6.0.3**  still include the dependency on **url_launcher_web 2.0.7**  breaking the building if you try to use url_launcher 6.0.3 on a project that uses Flutter 2.5.3
```
url_launcher_web:
    dependency: transitive
    description:
      name: url_launcher_web
      url: "https://pub.dartlang.org"
    source: hosted
    version: "2.0.7"
```

**Comment by stuartmorgan** [comment]
It turns out there's a client-side component to retraction:
```
Package retraction was introduced in Dart 2.15. In pre-2.15 SDKs, the pub version solver ignores the retracted status.
```

So retraction fixed this for Flutter 2.8, but not Flutter 2.5. Since 2.5 is now two stable releases behind, I don't think that edge case warrants the work if doing two more releases of `url_launcher_web` (one to revert back to the 2.0.6 state, then another to go back to the 2.0.8 state). For people who really want to stay on Flutter 2.5 there's a simple workaround, which is to pin
```
url_lanucher_web: 2.0.6
```

in pubspec.yaml. (Normal we don't recommend pinning since it prevents future updates, but url_launcher_web has dropped support for Flutter 2.5 so future updates to it wouldn't resolve for 2.5 anyway.)

@ditman @hterkelsen If you disagree, feel free to re-open and do the PRs.

**Comment by ditman** [comment]
@c-lucera-pvotal can you pin/override the following dependency?

```yaml
  url_launcher_web: 2.0.6
```

You'll need that until you move from Flutter 2.5 to something more modern, as @stuartmorgan mentioned [above](https://github.com/flutter/flutter/issues/97796#issuecomment-1031542721).

Apologies for the inconvenience! We weren't aware that pub package retraction had a client-side component too when we used it to solve this issue!
